create_derivatives/routines.py

```diff
@@ -37,14 +36,12 @@ class BaseRoutine(object):
     """
 
     def run(self):
-        object_list = []
-        for bag in Bag.objects.filter(process_status=self.start_process_status):
-            self.process_bag(bag)
-            bag.process_status = self.end_process_status
-            bag.save()
-            object_list.append(bag.bag_identifier)
-        msg = self.success_message if len(object_list) else self.idle_message
-        return msg, object_list
+        bag = Bag.objects.filter(process_status=self.start_process_status).first()
+        self.process_bag(bag)
```

Comment:
I think you need to check here that `bag` is truthy, because `first()` can return `None`: https://docs.djangoproject.com/en/3.2/ref/models/querysets/#first